c2corg_api/views/user.py

```diff
@@ -67,15 +67,18 @@ def validate_json_password(request):
         request.errors.add('body', 'password', 'Invalid')
 
 
+def available_user_attribute(attrname, value):
+    attr = getattr(User, attrname)
+    return DBSession.query(User).filter(attr == value).count() == 0
```

Comment:
I haven't done any performance tests. But `exists` stops as soon as the first result is found. `count` checks all rows/the whole index.